routes/owner.js

```diff
@@ -0,0 +1,101 @@
+const mongoose = require('mongoose')
+const express = require('express')
+const { Hotel, validate } = require('../models/hotel')
+const { Reservation } = require('../schemas/reservationSchema')
+const router = express.Router()
+
+// TODO: add auth middlewear, when it will be ready
+
+router.get('/hotels', async (req, res) => {
+  const hotel = await Hotel.find()
+
+  res.status(200).send(hotel)
+})
+
+router.post('/hotel', async (req, res) => {
+  const { error } = validate(req.body)
+  if (error) return res.status(400).send(error.details[0].message)
+
+  let hotel = new Hotel({
+    ownerId: req.body.ownerId,
+    localization: req.body.localization,
+    phoneNumber: req.body.phoneNumber,
+    name: req.body.name,
+    clientsRates: null,
+    email: req.body.email,
+    description: req.body.description,
+    rooms: [req.body.rooms],
+  })
+
+  await hotel.save()
+
+  res.status(200).send(hotel)
+})
+
+router.put('/hotel/:id', async (req, res) => {
+  const { error } = validate(req.body)
+  if (error) return res.status(400).send(error.details[0].message)
+
+  const hotel = await Hotel.findByIdAndUpdate(req.params.id, {
+    ownerId: req.body.ownerId,
+    localization: req.body.localization,
+    phoneNumber: req.body.phoneNumber,
+    name: req.body.name,
+    clientsRates: null,
+    email: req.body.email,
+    description: req.body.description,
+    rooms: [req.body.rooms],
+  })
+
+  if (!hotel) {
+    return res.status(404).send('Hotel with given ID was not found')
+  }
+  res.status.send(hotel)
+})
+
+router.delete('/hotel/:id', async (req, res) => {
+  const id = req.params.id
+  try {
+    const reservation = await Reservation.find({ hotelId: id })
+
+    if (reservation) return res.status(409).send('Remove reservations')
```

Comment:
My bad. It will be an array. Check it's length and rename reservation to reservations. Change message to: "Unable to remove hotel with provided ID. There are reservations connected to it." Status should be 400